rules/library.bzl

```diff
@@ -767,23 +776,33 @@ def apple_library(
         )
         import_vfsoverlays.append(import_name + "_vfs")
 
+    vendored_deps_contains_arm64_sim_slice = {}
     for xcframework in kwargs.pop("vendored_xcframeworks", []):
         xcframework_name, xcframework_name_vfs = _xcframework(library_name = name, **xcframework)
+        vendored_deps_contains_arm64_sim_slice[xcframework_name] = _xcframework_contains_arm64_sim_slice(xcframework)
         vendored_deps.append(xcframework_name)
         import_vfsoverlays.append(xcframework_name_vfs)
 
     # We don't currently support dynamic library
     for vendored_dynamic_library in kwargs.pop("vendored_dynamic_libraries", []):
         fail("no support for dynamic library: %s" % vendored_dynamic_library)
 
-    # TODO(jmarino)Perhaps it uses a import_middleman here
-    if len(vendored_deps):
-        import_middleman(name = name + ".import_middleman", deps = vendored_deps, tags = ["manual"])
+    vendored_deps_non_arm64_sim = [d for d in vendored_deps if not vendored_deps_contains_arm64_sim_slice.get(d, False)]
+    if vendored_deps_non_arm64_sim:
+        import_middleman(
+            name = name + ".import_middleman",
+            deps = vendored_deps_non_arm64_sim,
+            tags = ["manual"],
+        )
         deps += select({
             "@build_bazel_rules_ios//:arm64_simulator_use_device_deps": [name + ".import_middleman"],
-            "//conditions:default": vendored_deps,
+            "//conditions:default": vendored_deps_non_arm64_sim,
```

Comment:
My understanding is that even if the correct alias is picked up in `import_middleman` [is_sim_arm64](https://github.com/bazel-ios/rules_ios/blob/master/rules/import_middleman.bzl#L183) is still `True` making it run `update_in_place`

It's not clear to me how these two checks are supposed to work together and why I don't see the expected behaviour without the changes in this PR. Some help to clarify that would be great.

Reply:
Ok let me followup with you later on this - I don't fully understand why we need this either

Reply:
Use case: Framework `A` depends on xcframework `B` and we want to use SwiftUI Previews in `A`.

To achieve that we necessarily need `link_dynamic = True` in `A` (SwiftUI Previews requires dynamic linking). The issue is what to do with `B`.

What I've tried:

1. Set `link_dynamic = True` in `B`. This will bundle `B.framework` but the symbols table for `B.framework/B` is empty instead of containing the correctly symbols for the current slice being used (the binary created under the `import_middleman` path is correct but it's not the one used to create `B.framework/B`). This makes SwiftUI Previews fail due to missing `arm64` symbols. Had I found a fix for this I'm not sure if I'd still hit the issue described in `2.` next.
2. Leave `link_dynamic = False` in `B` when I realized `1.` might not be necessary since `A` is the module launching previews in Xcode so as long as `A` and it's public symbols are being linked dynamically seems like previews work fine based on my testing. The issue here is that the [update_in_place](https://github.com/bazel-ios/rules_ios/blob/master/tools/m1_utils/update_dynamic.sh#L23-L29) logic changes the signature causing SwiftUI Previews to crash:
```sh
Exception Type:  EXC_BAD_ACCESS (SIGKILL (Code Signature Invalid))
```

This PR only solves this partially as it will use the `arm64` + `sim` if it exists, bypassing `update_in_place`. If the xcframework contains only a non-simulator `arm64` slice then `update_in_place` will still run and crash SwiftUI Previews. 

The intent is to unblock SwiftUI Previews for scenarios where `arm64` + `sim` slices are present and investigate `1.` and `2.` in a follow up.